robotoff/app/core.py

```diff
@@ -78,24 +99,34 @@ def get_insights(
 
     if order_by is not None:
         if order_by == "random":
-            query = query.order_by(peewee.fn.Random())
+            query = query.order_by((peewee.fn.Random() * ProductInsight.n_votes).desc())
 
         elif order_by == "popularity":
             query = query.order_by(ProductInsight.unique_scans_n.desc())
 
+        elif order_by == "n_votes":
+            query = query.order_by(ProductInsight.n_votes.desc())
+
     if as_dict:
         query = query.dicts()
 
     return query.iterator()
 
 
-def save_insight(
+def save_annotation(
     insight_id: str,
     annotation: int,
+    device_id: str,
     update: bool = True,
     data: Optional[Dict] = None,
     auth: Optional[OFFAuthentication] = None,
+    verify_annotation: bool = False,
```

Comment:
Opinion: I find the name a bit misleading. maybe verified_annotation (True meaning we do not use voting) could be more explicit. Or trusted_annontation.

Reply:
I also found it a bit confusing, but was happy with the comment in the beginning of `save_annotation`. I defer to @kulizhsy for this discussion.

Reply:
I like Alex's suggestion of the name `verified_annotation` & flipping of the default.

Reply:
@kulizhsy will you change it or should I?

Reply:
Sure, let's go with `trusted_annotator`, I'll update the code.